terminology-update

LakinS opened the pull request:
Contains all 3 branches of changes, all of which passed X86 tests, haven't checked others yet, on their own, but start failing some tests after merging.

timothy-hayes committed:
arch-arm: Transactional Memory Extension (TME)

This patch extends the generic hardware transactional memory support in
Ruby and the O3/TimingSimpleCPU cores with the Arm-specific hardware
transactional memory architectural extensions (TME).

JIRA: https://gem5.atlassian.net/browse/GEM5-588

Change-Id: I8c663da977ed3e8c94635fcb11834bd001e92054
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/30329
Reviewed-by: Giacomo Travaglini <giacomo.travaglini@arm.com>
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>

mwboyer committed:
mem-ruby: Check number of vnets when creating links

Added error checking to ensure that the system has sufficient virtual
networks when setting latency and weight values.

Change-Id: I1b28144bbe9fefab0c0a6227f1fdf4ea10403061
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/32603
Tested-by: kokoro <noreply+kokoro@google.com>
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>

ebrickey committed:
cpu-o3: convert commit to new style stats

Change-Id: I859fe753d1a2ec2da8a4209d1db122f1014af5d6
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33315
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

hnpl committed:
base,misc: Add group definitions for newly tagged API in src/base

Signed-off-by: Hoa Nguyen <hoanguyen@ucdavis.edu>
Change-Id: If2f5ce3bc4f5d0a8cc31def17702223a27e6970e
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33034
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

andysan committed:
sim: Expose the system's byte order as a param

There are cases where a system's byte order isn't well-defined from an
ISA. For example, Arm implementations can be either big or little
endian, sometimes depending on a boot parameter. Decouple the CPU byte
order from the System's default byte order by exposing the System's
byte order as a parameter that defaults to big endian for SPARC and
POWER and little endian for everything else.

Change-Id: I24f87ea3a61b05042ede20dea6bb056af071d2c0
Signed-off-by: Andreas Sandberg <andreas.sandberg@arm.com>
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33175
Tested-by: kokoro <noreply+kokoro@google.com>
Reviewed-by: Gabe Black <gabeblack@google.com>

ebrickey committed:
cpu-o3: convert decode to new style stats

Change-Id: Ia67a51f3b2c2d40d8bf09f1636c721550f5e9a23
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33316
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Reviewed-by: Andreas Sandberg <andreas.sandberg@arm.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

shparekh00 committed:
misc: Replaced master/slave terminology

Change-Id: I4df2557c71e38cc4e3a485b0e590e85eb45de8b6

ebrickey committed:
cpu-o3: convert lsq_unit to new style stats

Removes unused stats: invAddrLoads, invAddrSwpfs, lsqBlockedLoads

Change-Id: Icd7fc6d8a040f4a1f9b190409b7cdb0a57fd68cf
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33394
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

ebrickey committed:
cpu-o3: convert rename to new style stats

Change-Id: Id34a85e40ad7e83d5805a034df6e0c5ad9b9af82
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33397
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

ebrickey committed:
cpu-o3: convert rob to new style stats

Change-Id: I84430d50c49742cd536dd75ce25184c2316dce51
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33398
Reviewed-by: Andreas Sandberg <andreas.sandberg@arm.com>
Maintainer: Andreas Sandberg <andreas.sandberg@arm.com>
Tested-by: kokoro <noreply+kokoro@google.com>

ebrickey committed:
cpu-o3: convert elastic trace to new style stats

Change-Id: If767f17b905a77e12058022a9e8bc65b854978a4
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33399
Reviewed-by: Andreas Sandberg <andreas.sandberg@arm.com>
Maintainer: Andreas Sandberg <andreas.sandberg@arm.com>
Tested-by: kokoro <noreply+kokoro@google.com>

hnpl committed:
ext: Force testlib to only create one Log object

Log object should remain being a singleton throughout the program.
The current code creates multiple Log objects, which at least
causes the issues of missing outputs in stdout.

E.g., "Logging call to command", which logs which command is
being called in a subprocess, is missing from stdout.

Change-Id: I96c5dd79c4f14e0a013c15d42d202397488d56b6
Signed-off-by: Hoa Nguyen <hoanguyen@ucdavis.edu>
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33715
Tested-by: kokoro <noreply+kokoro@google.com>
Reviewed-by: Bobby R. Bruce <bbruce@ucdavis.edu>
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>

eavivi4 committed:
cpu: convert trace cpu to new style stats

This required making minor changes to how the name was set for the
generators within the trace CPU to enable the stats to keep similar
names.

Change-Id: I9f97d4006a0edbd717fc34d0033b9548011d1631
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33875
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>
Reviewed-by: Ayaz Akram <yazakram@ucdavis.edu>

eavivi4 committed:
cpu-minor: convert fetch2 to new style stats

Change-Id: Idfe0f1f256c93209fe51140b9cab3b454153c597
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33975
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

powerjg commented on src/cpu/BaseCPU.py:
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
-        self.icache_port = ic.cpu_side
-        self.dcache_port = dc.cpu_side
-        self._cached_ports = ['icache.mem_side', 'dcache.mem_side']
+        self.icache_port = ic.cpu_side_port
```
This should just be cpu_side

powerjg commented on src/arch/arm/table_walker.cc:
```diff
@@ -55,7 +55,7 @@ using namespace ArmISA;
 TableWalker::TableWalker(const Params *p)
     : ClockedObject(p),
-      stage2Mmu(NULL), port(NULL), masterId(Request::invldMasterId),
+      stage2Mmu(NULL), port(NULL), _id(Request::invldUniqueId),
```
invldRequestorId?

powerjg commented on src/arch/arm/table_walker.cc:
```diff
@@ -97,11 +97,11 @@ TableWalker::~TableWalker()
 
 void
-TableWalker::setMMU(Stage2MMU *m, MasterID master_id)
+TableWalker::setMMU(Stage2MMU *m, RequestorID unique_id)
```
requestor_id?

powerjg commented on src/arch/arm/table_walker.hh:
```diff
@@ -829,8 +829,8 @@ class TableWalker : public ClockedObject
-    /** Master id assigned by the MMU. */
-    MasterID masterId;
+    /** Unique id assigned by the MMU. */
+    RequestorID _id;
```
_requestorId?

powerjg commented on src/arch/arm/tlb.cc:
```diff
@@ -107,10 +107,10 @@ TLB::init()
 {
     stage2Mmu = m;
-    tableWalker->setMMU(m, master_id);
+    tableWalker->setMMU(m, unique_id);
```
requestor_id?

powerjg commented on src/arch/arm/tlb.hh:
```diff
@@ -224,7 +224,7 @@ class TLB : public BaseTLB
     TableWalker *getTableWalker() { return tableWalker; }
 
-    void setMMU(Stage2MMU *m, MasterID master_id);
+    void setMMU(Stage2MMU *m, RequestorID unique_id);
```
requestor_id?

powerjg commented on src/arch/arm/tracers/tarmac_parser.cc:
```diff
@@ -1287,7 +1287,7 @@ TarmacParserRecord::readMemNoEffect(Addr addr, uint8_t *data, unsigned size,
 
     req->setVirt(addr, size, flags, thread->pcState().instAddr(),
-                 Request::funcMasterId);
+                 Request::funcUniqueId);
```
funcRequestorId

powerjg commented on src/arch/x86/pagetable_walker.cc:
```diff
@@ -519,7 +519,7 @@ Walker::WalkerState::stepWalk(PacketPtr &write)
         flags.set(Request::UNCACHEABLE, uncacheable);
         RequestPtr request = std::make_shared<Request>(
-            nextRead, oldRead->getSize(), flags, walker->masterId);
+            nextRead, oldRead->getSize(), flags, walker->_id);
```
_requestorId?

powerjg commented on src/arch/x86/pagetable_walker.hh:
```diff
@@ -168,7 +168,7 @@ namespace X86ISA
         TLB * tlb;
         System * sys;
-        MasterID masterId;
+        RequestorID _id;
```
_requestorId?

powerjg commented on src/arch/x86/pagetable_walker.hh:
```diff
@@ -204,7 +204,7 @@ namespace X86ISA
             ClockedObject(params), port(name() + ".port", this),
             funcState(this, NULL, NULL, true), tlb(NULL), sys(params->system),
-            masterId(sys->getMasterId(this)),
+            _id(sys->getUniqueId(this)),
```
getRequestorId?

powerjg commented on src/cpu/BaseCPU.py:
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
-        self.dcache_port = dc.cpu_side
-        self._cached_ports = ['icache.mem_side', 'dcache.mem_side']
+        self.icache_port = ic.cpu_side_port
+        self.dcache_port = dc.cpu_side_port
```
ditto, cpu_side

powerjg commented on src/cpu/BaseCPU.py:
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
-        self._cached_ports = ['icache.mem_side', 'dcache.mem_side']
+        self.icache_port = ic.cpu_side_port
+        self.dcache_port = dc.cpu_side_port
+        self._cached_ports = ['icache.mem_side_port', 'dcache.mem_side_port']
```
should be mem_side for both

powerjg commented on src/cpu/BaseCPU.py:
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
-                self.dtb.walker.port = dwc.cpu_side
-                self._cached_ports += ["itb_walker_cache.mem_side", \
-                                       "dtb_walker_cache.mem_side"]
+                self.itb.walker.port = iwc.cpu_side_port
```
cpu_side

powerjg commented on src/cpu/BaseCPU.py:
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
-                self._cached_ports += ["itb_walker_cache.mem_side", \
-                                       "dtb_walker_cache.mem_side"]
+                self.itb.walker.port = iwc.cpu_side_port
+                self.dtb.walker.port = dwc.cpu_side_port
```
cpu_side

powerjg commented on src/cpu/BaseCPU.py:
```diff
@@ -211,17 +211,17 @@ def connectAllPorts(self, cached_bus, uncached_bus = None):
-                                       "dtb_walker_cache.mem_side"]
+                self.itb.walker.port = iwc.cpu_side_port
+                self.dtb.walker.port = dwc.cpu_side_port
+                self._cached_ports += ["itb_walker_cache.mem_side_port", \
```
Both of these should be mem_side

powerjg commented on src/cpu/BaseCPU.py:
```diff
@@ -237,8 +237,8 @@ def addTwoLevelCacheHierarchy(self, ic, dc, l2c, iwc=None, dwc=None,
         self.l2cache = l2c
-        self.toL2Bus.master = self.l2cache.cpu_side
-        self._cached_ports = ['l2cache.mem_side']
+        self.toL2Bus.mem_side_port = self.l2cache.cpu_side_port
```
mem_side_port is correct, but it should be cpu_side

powerjg commented on src/cpu/base.cc:
```diff
@@ -122,8 +122,8 @@ CPUProgressEvent::description() const
     : ClockedObject(p), instCnt(0), _cpuId(p->cpu_id), _socketId(p->socket_id),
-      _instMasterId(p->system->getMasterId(this, "inst")),
-      _dataMasterId(p->system->getMasterId(this, "data")),
+      _instUniqueId(p->system->getUniqueId(this, "inst")),
```
RequestorId?

powerjg commented on src/cpu/base.cc:
```diff
@@ -250,7 +250,7 @@ BaseCPU::mwaitAtomic(ThreadID tid, ThreadContext *tc, BaseTLB *dtb)
         size = secondAddr - addr;
 
-    req->setVirt(addr, size, 0x0, dataMasterId(), tc->instAddr());
+    req->setVirt(addr, size, 0x0, dataUniqueId(), tc->instAddr());
```
dataRequestorId?

powerjg commented on src/cpu/base.hh:
```diff
@@ -124,10 +124,10 @@ class BaseCPU : public ClockedObject
 
     /** instruction side request id that must be placed in all requests */
-    MasterID _instMasterId;
+    RequestorID _instUniqueId;
```
requestorId?

powerjg commented on src/arch/arm/stage2_mmu.cc:
```diff
@@ -49,13 +49,13 @@ using namespace ArmISA;
     : SimObject(p), _stage1Tlb(p->tlb), _stage2Tlb(p->stage2_tlb),
       port(_stage1Tlb->getTableWalker(), p->sys),
-      masterId(p->sys->getMasterId(_stage1Tlb->getTableWalker()))
+      _id(p->sys->getUniqueId(_stage1Tlb->getTableWalker()))
```
getRequestorId

powerjg commented on src/arch/arm/stage2_mmu.hh:
```diff
@@ -60,7 +60,7 @@ class Stage2MMU : public SimObject
 
     /** Request id for requests generated by this MMU */
-    MasterID masterId;
+    RequestorID _id;
```
I think Andreas wanted this to be _requestorId

powerjg commented on src/arch/riscv/pagetable_walker.cc:
```diff
@@ -396,7 +396,7 @@ Walker::WalkerState::stepWalk(PacketPtr &write)
         //If we didn't return, we're setting up another read.
         RequestPtr request = std::make_shared<Request>(
-            nextRead, oldRead->getSize(), flags, walker->masterId);
+            nextRead, oldRead->getSize(), flags, walker->_id);
```
_requestorId?

powerjg commented on src/cpu/base.hh:
```diff
@@ -182,9 +182,9 @@ class BaseCPU : public ClockedObject
 
     /** Reads this CPU's unique data requestor ID */
-    MasterID dataMasterId() const { return _dataMasterId; }
+    RequestorID dataUniqueId() const { return _dataUniqueId; }
```
requestorId

powerjg commented on src/cpu/checker/cpu.hh:
```diff
@@ -89,7 +89,7 @@ class CheckerCPU : public BaseCPU, public ExecContext
 
     /** id attached to all issued requests */
-    MasterID masterId;
+    RequestorID _id;
```
_requestorId?

powerjg commented on src/cpu/testers/traffic_gen/base.hh:
```diff
@@ -53,7 +53,7 @@ class System;
 
 /**
- * The traffic generator is a master module that generates stimuli for
+ * The traffic generator is a memside module that generates stimuli for
```
Just "module"

powerjg commented on src/cpu/testers/traffic_gen/base_gen.cc:
```diff
@@ -46,8 +46,8 @@
 
-BaseGen::BaseGen(SimObject &obj, MasterID master_id, Tick _duration)
-    : _name(obj.name()), masterID(master_id),
+BaseGen::BaseGen(SimObject &obj, RequestorID unique_id, Tick _duration)
```
requestor_id

powerjg commented on src/cpu/testers/traffic_gen/base_gen.cc:
```diff
@@ -69,19 +69,19 @@ BaseGen::getPacket(Addr addr, unsigned size, const MemCmd& cmd,
 
 StochasticGen::StochasticGen(SimObject &obj,
-                             MasterID master_id, Tick _duration,
+                             RequestorID unique_id, Tick _duration,
```
requestor_id

powerjg commented on src/cpu/testers/traffic_gen/base_gen.hh:
```diff
@@ -85,10 +85,10 @@ class BaseGen
      *
      * @param obj simobject owning the generator
-     * @param master_id MasterID set on each request
+     * @param unique_id RequestorID set on each request
```
requestor_id

powerjg commented on src/cpu/testers/traffic_gen/base_gen.hh:
```diff
@@ -85,10 +85,10 @@ class BaseGen
      * @param _duration duration of this state before transitioning
      */
-    BaseGen(SimObject &obj, MasterID master_id, Tick _duration);
+    BaseGen(SimObject &obj, RequestorID unique_id, Tick _duration);
```
requestor_id

powerjg commented on src/cpu/testers/traffic_gen/traffic_gen.hh:
```diff
@@ -45,15 +45,15 @@
 
 /**
- * The traffic generator is a master module that generates stimuli for
+ * The traffic generator is a memside module that generates stimuli for
```
just "module"

hnpl committed:
ext: Add post_test_procedure to testlib runner

This procedure is trigger after a test has finished and before
the tearing down process kicks in.

Change-Id: I58ce10814fbc80d96f2f72565491b18de0ec290a
Signed-off-by: Hoa Nguyen <hoanguyen@ucdavis.edu>
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33997
Tested-by: kokoro <noreply+kokoro@google.com>
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>

hnpl committed:
ext,tests: Copy test's output files from /tmp to testing-results

When a test is complete, the output files are in a random folder
in /tmp.

This commit adds a procedure copying those files to
testing-results/SuiteUID/TestUID/ folder, where SuiteUID and
TestUID are the corresponding uid's of the test.

This procedure is triggered after a test is complete and before
the folder in /tmp being removed.

Change-Id: Id960e7f2f1629769008ae99aff4c8bfafa9ca849
Signed-off-by: Hoa Nguyen <hoanguyen@ucdavis.edu>
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/33998
Tested-by: kokoro <noreply+kokoro@google.com>
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>

LakinS commented on src/mem/cache/base.hh:
```diff
@@ -572,7 +572,7 @@ class BaseCache : public ClockedObject
      * @param pkt The request to perform.
      * @param fromCpuSide from the CPU side port or the memory side port
      */
-    virtual void functionalAccess(PacketPtr pkt, bool from_cpu_side);
```
Should this be switched back?

powerjg replied:
Yeah, I'd switch it back

gabemblack committed:
arm: Remove "using namespace ArmISA" from arch/arm/isa_traits.hh.

This has been in this file since it was created in 2009. No global "using
namespace ${NAMESPACE}" should ever appear in a .hh file since then that
namespace is "used" in all files that include the .hh, even if they
aren't aware of it or even actively don't want to.

Change-Id: Idb7d7c5b959077eb4905fbb2044aa55959b8f37f
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/34155
Reviewed-by: Andreas Sandberg <andreas.sandberg@arm.com>
Maintainer: Andreas Sandberg <andreas.sandberg@arm.com>
Tested-by: kokoro <noreply+kokoro@google.com>

gabemblack committed:
arm: Replicate the PageBytes constant in the ArmSystem class.

When isa_traits.hh hopefully goes away in the not too distant future,
this constant will need somewhere to live so ARM components can find it.
There are valid arguments that this should not be a constant in the
first place, but that's outside the scope of this change.

Change-Id: Ic5bd046dc1cc196b3cf6b6c36878fdbf5eb4c0bf
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/34170
Reviewed-by: Andreas Sandberg <andreas.sandberg@arm.com>
Maintainer: Andreas Sandberg <andreas.sandberg@arm.com>
Tested-by: kokoro <noreply+kokoro@google.com>

gabemblack committed:
dev,arm: Use the ArmSystem::PageBytes constant in the generic timer.

This component very specific to ARM, and so there's no reason to use
generic interfaces to get the page size.

Change-Id: Id757b5742c807c5f616a6dc8df94a7709932d071
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/34171
Reviewed-by: Andreas Sandberg <andreas.sandberg@arm.com>
Maintainer: Andreas Sandberg <andreas.sandberg@arm.com>
Tested-by: kokoro <noreply+kokoro@google.com>

gabemblack committed:
arch: Add a virtual destructor to BaseHTMCheckpoint.

Since it has virtual methods, it should also have a virtual destructor.
My compiler complains otherwise, which breaks my build.

Change-Id: I44bba97b76931bab6e3511fcdee79831080c12d4
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/34215
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

gabemblack committed:
cpu: Fix style and add overrides to bas_dyn_inst.hh.

Either return types, brackets and the function body should all be on
their own line, or the entire function should be on a single line.

Consistently place the * or & up against the variable name and not the
type name. There isn't an official rule for which to use, but the
majority of existing uses were this way.

Add overrides for overridden virtual methods.

These fixes get rid of compiler warnings which are breaking the build
for me.

Change-Id: Ifc6ace4794a66ffd031ee686f6b6ef888004d786
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/34216
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

gabemblack committed:
mem: Remove the unused nvm private member from NVMInterface::Rank.

This unused (and otherwise unusable) member caused a compiler warning
and broke the build for me. It can be reintroduced if used in the
future.

Change-Id: I48181f6bca60c059e74727290950adfb9a194680
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/34217
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>

giactra committed:
arch-arm: Fix ArmISA namespace requirement for TME instructions

This is needed after:

https://gem5-review.googlesource.com/c/public/gem5/+/34155

Change-Id: I8ef0b5ce9cd5ae5224331e1c9347fdd9e884a536
Signed-off-by: Giacomo Travaglini <giacomo.travaglini@arm.com>
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/34235
Reviewed-by: Andreas Sandberg <andreas.sandberg@arm.com>
Maintainer: Andreas Sandberg <andreas.sandberg@arm.com>
Tested-by: kokoro <noreply+kokoro@google.com>
